### ISSUE TITLE
Notification and Date Pannel are outside of the screen

### ISSUE THREAD

**Issue by JulianTurner** [Jas-SinghFSU/HyprPanel#28]
**Describe the bug**
The panel is cut off and outside of the screen

**To Reproduce**
open any panel that goes to the right side

**Expected behavior**
The right side of the display is set wrong

**Screenshots**
![image](https://github.com/user-attachments/assets/ed4d9976-6714-4393-8b2d-8d5e97cf46aa)
![image](https://github.com/user-attachments/assets/a3a1ac6c-e57a-4624-81f2-c88bcd352ced)

**Desktop (please complete the following information):**
 - Distribution: [e.g. Arch Linux] Arch Linux
 - Window Manager/Desktop Environment: [e.g. Hyprland] Hyprland

**Additional context**
I have rebooted the system and only problems on the right side

**Comment by Jas-SinghFSU** [comment]
Could you provide your screen resolution and any scaling/resolution related config that you have in your hyprland.conf?

And the output of `echo $GDK_SCALE` if you have any custom scaling.

**Comment by smathles** [comment]
I can replicate this, only I additionally have my "main menu" (top left arch logo button) also cut off at the bottom of my screen, left/right position is perfect. 

I have no `$GTK_SCALING` parameter set, BUT I use plasma 6 side-by-side with hyprland and have plasma set to 150% zoom as I have a HiDPI monitor (1920x1080, scaling as if it was 1280x720). The only scaling parameter in `hyprland.conf` as far as I can see is for `XCURSOR_SIZE=24`, as the mouse was not scaled by default.
![start_too_big](https://github.com/user-attachments/assets/970dbcb5-df99-4ea7-aa82-2150976b3b33)
![notifications_too_big](https://github.com/user-attachments/assets/47862fd6-8514-400f-9cfe-627f92d2b393)

**Comment by dringwald** [comment]
I'm getting the same. I tinkered around with the code but couldn't find anything. I put in some console.log()'s at DropdownMenu.ts and got the below output when opening the Dashboard on the left, vs the Notifications on the right. ("MarginRight/Left 1" is called right after being assigned, and "MarginRight/Left 2" is right after the "self.set_margin" calls.

Opening Dashboard
```
Gjs-Console-Message: 00:50:29.565: marginRight 1: 2322.5 
Gjs-Console-Message: 00:50:29.565: marginLeft 1: -147.5
Gjs-Console-Message: 00:50:29.566: gdkScale: 
Gjs-Console-Message: 00:50:29.566: value[0]: 45
Gjs-Console-Message: 00:50:29.566: monWidth: 2560
Gjs-Console-Message: 00:50:29.566: currentWidth: 385
Gjs-Console-Message: 00:50:29.566: marginRight 2: 2175
Gjs-Console-Message: 00:50:29.566: marginLeft 2: 0
```

Opening Notifications:
```
Gjs-Console-Message: 00:50:36.832: marginRight 1: 862.5
Gjs-Console-Message: 00:50:36.832: marginLeft 1: 1386.5
Gjs-Console-Message: 00:50:36.832: gdkScale: 
Gjs-Console-Message: 00:50:36.832: value[0]: 1542
Gjs-Console-Message: 00:50:36.832: monWidth: 2560
Gjs-Console-Message: 00:50:36.832: currentWidth: 311
Gjs-Console-Message: 00:50:36.832: marginRight 2: 862.5
Gjs-Console-Message: 00:50:36.833: marginLeft 2: 1386.5
```

I can see that the out-of-bounds logic is working for the Dashboard as it corrects the negative margin-left, but the same doesn't work for margin-right. 

NixOS if it matters.

**Comment by Jas-SinghFSU** [comment]
@dringwald Thank for all that info, it's greatly appreciated. I'll have to look into why this is happening. If you wouldn't mind, can you log the global mouse position value?
You can do this by dropping this `console.log(value)` anywhere within the `globalMousePos.connect` scope. 

The margins on both are being calculated properly, but I'm guessing that the position of the mouse click is being calculated incorrect so the correct margins are being applied on the wrong coordinates.

**Comment by JulianTurner** [comment]
> echo $GDK_SCALE

none, its empty

**Comment by dringwald** [comment]
@Jas-SinghFSU 
`console.log(value)` gets me 1542,20. 

A `console.log(clickPos)` in modules/bar/utils.ts in `openMenu` gets me 1547,20. So nothing too different there.

Logging `value` after clicking on the media module (middle) and dashboard (left) gets me 800, 19 and 45, 24 respectively. So it seems to be scaling appropriately, it just thinks the monitor is ~1600 pixels wide.

**Comment by Jas-SinghFSU** [comment]
Yeah, that's really odd... the click coordinate being gathered is assuming a different screen width than the margin setting logic...

**Comment by Jas-SinghFSU** [comment]
@JulianTurner @dringwald @smathles Could you pull the branch `menu_test ` and let me know if that had any change?
Instead of using `event.get_root_coords()`, I instead get the cursor position using `hyprctl cursorpos`; this may hopefully fix it.

**Comment by JulianTurner** [comment]
@Jas-SinghFSU  Hey, the issue still persist even on the new branch.

**Comment by smathles** [comment]
@Jas-SinghFSU I have the same issue as Julian, no change in this branch unfortunately. If I `export GDK_SCALE=1.6`, I do have the menus on the right hand side of the screen successfully be inside the screen, but they're still scaled much too big and I cannot see the bottom of the menus (also true for the scrollable elements inside the settings widget, they "finish" scrolling before I can actually see the bottom).

So the GDK_SCALE does indeed make the notification + date panel fall inside the screen on the right hand side, but it still doesn't affect the menu's scaling, which may or may not be the same problem.

**Comment by smathles** [comment]
Update: I tried Julian's solution  of setting `monitor=,preferred,auto,auto` to `monitor=,preferred,auto,1` to set the scaling of the monitor to 1 instead of hyprland's automatic scaling of 1.5 (didn't realise hyprland was clever enough to sense HiDPI monitor and scale accordingly, learn something new everyday I guess). When scaled to 100% instead of the automatic 150%, it works (with the exception of the icons + spacing still being slightly off-center)! 

![offset_icons_2](https://github.com/user-attachments/assets/f46766e4-f52a-45a9-acef-2a4c7d1b3bba)
![offset_icons](https://github.com/user-attachments/assets/692b849b-c6c1-4b42-9bdf-6d932db0480f)


The overall environment though is now a bit small for my liking, so IMO scaling for the ags widgets might be better? No idea how/if that could be implemented sorry.

**Comment by Jas-SinghFSU** [comment]
This is super useful information. Thank you for sharing that; I will see if I can replicate it and handle the positioning better.

**Comment by Jas-SinghFSU** [comment]
Update: I was able to replicate this give your awesome info. Debugging now!

**Comment by Jas-SinghFSU** [comment]
@JulianTurner @dringwald @smathles Could you please git pull and verify if this Merge fixed your issue?

**Comment by smathles** [comment]
@Jas-SinghFSU Unfortunately it hasn't, at least not on the main branch. I have not set gdk_scaling to anything, menus still appear the same size as before (150%, desired 100% unlike the system environment).

**Comment by Jas-SinghFSU** [comment]
@smathles Could you run `hyprctl monitors` and give me your output please?

Additionally if you could run `git show` and provide the commit id as well.

**Comment by smathles** [comment]
`hyprctl monitors`
```
Monitor eDP-1 (ID 0):
	1920x1080@60.01000 at 0x0
	description: InfoVision Optoelectronics (Kunshan) Co.Ltd China 0x057D
	make: InfoVision Optoelectronics (Kunshan) Co.,Ltd China
	model: 0x057D
	serial: 
	active workspace: 3 (3)
	special workspace: 0 ()
	reserved: 0 36 0 0
	scale: 1.50
	transform: 0
	focused: yes
	dpmsStatus: 1
	vrr: 0
	activelyTearing: false
	disabled: false
	currentFormat: XRGB8888
	availableModes: 1920x1080@60.01Hz
```
	
`git show`
```
commit 089395207ccf5ddb70faa008850bb6134a95447f (HEAD -> master, origin/master, origin/HEAD)
```

**Comment by smathles** [comment]
Though I just read the section in the README regarding scaling based off of font size, manually set that to 0.8 instead of 1.2 and it works fine! Had an initial teething issue where the notifications/clock widget were too far to the right/left, but it sorted itself out after a few seconds (probably still initialising).

Not sure if that's best practice, to change the font size vs. hyprland global scaling, but it worked great for me! Still a few offsets in the icons, might make a separate issue for that though (no offset issues on my non-HiDPI main pc).

**Comment by Jas-SinghFSU** [comment]
@smathles  Font size scaling your panel is very much intentional, hence why the default units are `em` and the entire application scales in such a way :). Happy that you've got that nicely scaled. As for the off-center fonts,  make sure that you're using a mono-space fonts; otherwise that just the downside of nerd-fonts since they use more than one cell unlike regular text :(.

How about the menu spawning? Is that still spawning your menus off screen or were you able to git pull the latest branch ad fix that aspect?

**Comment by smathles** [comment]
Menu spawning is perfect now, thanks!

Wasn't aware that mono-space fonts had a purpose like that, probably should have guessed from the name lol. I'll give that a go 👍.

**Comment by Jas-SinghFSU** [comment]
Awesome :smile: ! Happy we got it all sorted!

**Comment by JulianTurner** [comment]
@Jas-SinghFSU i alo can confirm it is working now, thanks buddy :-)

**Comment by Irubataru** [comment]
Don't mean to necro this issue, but I am still having this issue when I use hyprland autoscaling. In my hyprland config I have

```
monitor=eDP-1,highres,auto,1.33333
```

If I set the scaling to 1, everything works and is nice and aligned, but if I keep it at 1.33, it overflows.

I also use [kanshi](https://sr.ht/~emersion/kanshi/) to change config when I connect or disconnect monitors, that might do something. But I tried disabling kanshi, and rebooting with just hyprland setting the scaling, but that didn't help.
